test/integration/custom-routes-i18n-index-redirect/test/index.test.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-env jest */
+
+import url from 'url'
+import http from 'http'
+import { join } from 'path'
+import {
+  launchApp,
+  killApp,
+  findPort,
+  nextBuild,
+  nextStart,
+  File,
+  fetchViaHTTP,
+} from 'next-test-utils'
+
+const appDir = join(__dirname, '..')
+const nextConfig = new File(join(appDir, 'next.config.js'))
+let server
+let externalPort
+let appPort
+let app
+
+const runTests = () => {
+  it('should respond to default locale redirects correctly for index redirect', async () => {
+    for (const [path, dest] of [
+      ['/', '/destination'],
+      ['/en', '/destination'],
+      ['/fr', '/fr/destination'],
+    ]) {
+      const res = await fetchViaHTTP(appPort, path, undefined, {
+        redirect: 'manual',
+      })
```

Comment:
The tests are based on `custom-routes-i18n`:
https://github.com/vercel/next.js/blob/7bbd239fe7e6d72ef467484619b978c11f5fbb70/test/integration/custom-routes-i18n/test/index.test.js#L26-L33

Reply:
Confirmed that if I run this test on `canary` it would fail:

```
expect(received).toEqual(expected) // deep equality

  Expected: "/destination"
  Received: "/en/destination"

    36 |       if (dest) {
    37 |         const text = await res.text()
  > 38 |         expect(text).toEqual(dest)
       |                      ^
    39 |         if (dest.startsWith('/')) {
    40 |           const parsed = url.parse(res.headers.get('location'))
    41 |           expect(parsed.pathname).toBe(dest)

    at Object.<anonymous> (integration/custom-routes-i18n-index-redirect/test/index.test.js:38:22)
```